portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java

```diff
@@ -1962,9 +1959,6 @@ public DLFileEntry updateStatus(
 
 		// File entry
 
-		DLFileEntry dlFileEntry = dlFileEntryPersistence.findByPrimaryKey(
-			dlFileVersion.getFileEntryId());
```

Comment:
Hi @Preston-Crary,

I agree with other changes you've made in the form of `fileEntry = update(fileEntry...)`. It is more correct to use the new object than reuse the potentially stale one. However I don't fully understand why this method needs to change. We were not reusing any objects (it received just the ids) and the only operation that was being performed between the find (`dlFileEntryPersistence.findByPrimaryKey(`) and the update (`dlFileEntryPersistence.update(dlFileEntry);`) was the versions find. I would have thought that `dlFileVersionPersistence.findByF_S(`was to blame here and the solution should be applied inside it.

Please let us know your thoughts.

Thanks for your time!